photon-lib/src/main/scala/com/linkedin/photon/ml/data/scoring/CoordinateDataScores.scala

```diff
@@ -15,20 +15,21 @@
 package com.linkedin.photon.ml.data.scoring
 
 import org.apache.spark.rdd.RDD
-import org.apache.spark.rdd.RDD.rddToPairRDDFunctions
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.functions.{col, udf}
 
-import com.linkedin.photon.ml.Types.UniqueSampleId
-import com.linkedin.photon.ml.constants.MathConst
 import com.linkedin.photon.ml.data.GameDatum
+import com.linkedin.photon.ml.constants.{DataConst, MathConst}
 
 /**
  * The class used to track scored data points throughout training. The score objects are scores only, with no additional
  * information.
  *
- * @param scoresRdd The scores consist of (unique ID, score) pairs as explained above.
+ * @param scores The scores dataframe consist of (unique ID, score) pairs as explained above.
  */
-protected[ml] class CoordinateDataScores(override val scoresRdd: RDD[(UniqueSampleId, Double)])
-  extends DataScores[Double, CoordinateDataScores](scoresRdd) {
+protected[ml] class CoordinateDataScores(override val scores: DataFrame)
```

Comment:
This class should be deleted - this is not how the scoring was described in the doc. The scores should be computed as a new column onto the raw training data `DataFrame`, not a separate `DataFrame` that is merged with the raw training data.